sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -256,8 +256,19 @@ public abstract partial class Response : System.IDisposable
     public sealed partial class ResponseError
     {
         public ResponseError(string? code, string? message) { }
+        public ResponseError(string? code, string? message, string? target, Azure.ResponseInnerError? innerError = null, System.Collections.Generic.IReadOnlyList<Azure.ResponseError>? details = null) { }
```

Comment:
Do we need this constructor?  What if we had a Parse() method instead that took response.Content?

Reply:
This was intended only for testing. The idea was that someone could return it in, say, a mock. Making them parse it out of `BinaryData` (or whatever) seems unnecessary, but I'm not opposed to supporting it in general. Or do we just provide a constructor that takes a `BinaryData` and let `BinaryData` handle all the encoding/decoding work?

If it would help - though I'm also not entirely opposed to getting rid of this ctor - I could add in the comment this is intended for testing. I would've done this in a mock factory, except that Core doesn't have any mock factory currently.

@KrzysztofCwalina @tg-msft what are your thoughts here?